indy_node/server/batch_handlers/domain_batch_handler.py

```diff
@@ -0,0 +1,69 @@
+from collections import Callable
+
+import base58
+
+from plenum.server.batch_handlers.domain_batch_handler import DomainBatchHandler as PDomainBatchHandler
+from plenum.server.database_manager import DatabaseManager
+
+
+class DomainBatchHandler(PDomainBatchHandler):
+    def __init__(self, database_manager: DatabaseManager):
+        super().__init__(database_manager)
+        self.post_batch_creation_handlers = []
+        self.post_batch_commit_handlers = []
+        self.post_batch_rejection_handlers = []
+
+        self._add_default_handlers()
+
+    def commit_batch(self, txn_count, state_root, txn_root, pp_time):
+        result = super().commit_batch(txn_count, state_root, txn_root, pp_time)
+        self.post_batch_commited(state_root)
+        return result
+
+    def post_batch_applied(self, state_root):
+        for handler in self.post_batch_creation_handlers:
+            handler(state_root)
+
+    def post_batch_rejected(self):
+        for handler in self.post_batch_rejection_handlers:
+            handler()
+
+    def post_batch_commited(self, state_root):
+        for handler in self.post_batch_commit_handlers:
+            handler(state_root)
+
+    def _add_default_handlers(self):
```

Comment:
This needs to be a separate class `IdrCacheBatchHandler` that extends `BatchHandler`.

Reply:
fixed